eng/common/scripts/SemVer.ps1

```diff
@@ -19,7 +19,7 @@ class AzureEngSemanticVersion {
     [bool] $IsPrerelease
     [string] $RawVersion
     # Regex inspired but simplifie from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
-    static [string] $SEMVER_REGEX = "^(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:-?(?<prelabel>[a-zA-Z-]*)(?:\.?(?<prenumber>0|[1-9]\d*)))?$"
+    static [string] $SEMVER_REGEX = "(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:-?(?<prelabel>[a-zA-Z-]*)(?:\.?(?<prenumber>0|[1-9]\d*)))?"
```

Comment:
Since you removed the begin/end here you might want to add them in the ParseVersionString method below just so we can be sure we are still only parsing a version.